src/app/games/martian-chess/MartianChessMove.ts

```diff
@@ -0,0 +1,74 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { Direction } from 'src/app/jscaip/Direction';
+import { NumberEncoder } from 'src/app/jscaip/Encoder';
+import { MoveCoordToCoord } from 'src/app/jscaip/MoveCoordToCoord';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+
+export class MartianChessMoveFailure {
+
+    public static readonly START_COORD_OUT_OF_RANGE: string = 'Start coord cannot be out of range';
+
+    public static readonly END_COORD_OUT_OF_RANGE: string = 'End coord cannot be out of range';
+
+    public static readonly PAWN_MUST_MOVE_ONE_DIAGONAL_STEP: () => string = () => $localize`Pawns must move one diagonal step`;
```

Comment:
move *by* one diagonal step?

Reply:
C'est apropprié "by" ? J'trouve que ça fait tâche.

Reply:
alors je propose à la place: `Pawns must move one step diagonally`

Reply:
Les toasts on les fini par un point finalement, je sais, je demande ça pour la 742ème fois :D
Parce que si oui: tout doit finir par un point alors ?

Reply:
tous les toasts oui.
Tout non, pas les titres.
Tu veux qu'on fasse un document qui reprends ces règles ? :D

Reply:
ça pourrait être bien haha, "nos conventions: la charte de l'autisme" :D !

Reply:
Où est-ce qu'on met ça ? Dans `doc/conventions.md`?

Reply:
D'accord!

Reply:
ah bah y'a le ticket qui a été créé, complétons le d'abord puis on le copiera dans ce fichier je dirais !

Reply:
voilà j'ai noté "tout sauf les titres termine par un point" tu peux résoudre